### ISSUE TITLE
`Save layers` information is not always populated on prod.

### ISSUE THREAD

**Issue by DragaDoncila** [chanzuckerberg/napari-hub#721]
#### Description
I noticed [the preview of `napari-demo` ](https://preview.napari-hub.org/chanzuckerberg/napari-demo/19) has the `Save Layers` info populated, but [on prod](https://www.napari-hub.org/plugins/napari-demo) it says "information not submitted" (screenshots below).

#### Steps/Code to Reproduce
1. Opened PR to `napari-demo` pointing to branch with the new changes to `npe2 fetch`
2. Checked preview page, save layer information is populated
3. Checked `napari-demo` listing [on prod](https://www.napari-hub.org/plugins/napari-demo) and the information is not populated

Note - I've checked the metadata on prod and the `writer_save_layers` info is there at the backend, so I suspect this is a frontend issue. I had a quick look around and I noticed a `PluginWriterSaveLayer` enum that only has `image`, `points` and `shapes` in it, which is only a small subset of what the save layers info could hold. But this might be unrelated - @codemonkey800 any ideas?

#### Expected Results
I expected to see the information populated on both preview page and prod. Alternatively if we don't want to show that information on prod for whatever reason, we should hide the heading entirely.

##### Prod
![image](https://user-images.githubusercontent.com/17995243/199127504-26bf7413-e17a-4af2-b53f-6677f46591fc.png)

##### Preview
![image](https://user-images.githubusercontent.com/17995243/199127612-31e2b37c-5523-4915-a56a-50a9d61a2211.png)

**Comment by DragaDoncila** [comment]
A bit more context:

[napari-splineit](https://www.napari-hub.org/plugins/napari-splineit) does have info populated on prod, they just have `shapes` in their save layers metadata. [napari-deeplabcut](https://www.napari-hub.org/plugins/napari-deeplabcut) don't have info populated, but their metadata is:

```
  "writer_save_layers": [
    "points{1}",
    "shapes{1}"
  ]
``` 

Based on this I'd say we're being too strict on the data validation for this field in the frontend.

**Comment by DragaDoncila** [comment]
@codemonkey800 we don't do any additional parsing of the `writer_save_layers` field beyond reading them all into a set. It looks like the frontend is not expecting any[ quantifiers](https://napari.org/stable/plugins/guides.html#layer-type-constraints) to be present e.g. `image+` or `image{2}`. Is that desired behaviour? If so, I can strip the quantifiers in the backend. We would still need to add the other layer types to the list on the frontend though. Here's the full list:

- Image
- Labels
- Points
- Shapes
- Surface
- Tracks
- Vectors

**Comment by DragaDoncila** [comment]
My opinion would be yes be strict about layer types, and no don't show quantifiers - just show whether there's any writer that mentions that layer type - I think the exact quantifiers are too specific. Would also definitely need to remove them from the filters. But yes, curious to hear what the others think.

Also worth noting the manifest is already validated at the backend and an invalid manifest wouldn't be loaded, so you could relax the frontend validation.

**Comment by neuromusic** [comment]
+1 to @DragaDoncila's comments re: 1 & 2

**Comment by DragaDoncila** [comment]
Ok so it sounds like action items are:

- [ ] add remaining layer types to enum in the frontend.
- [ ] update `writer_save_layers` in backend to remove quantifiers.

**Comment by DragaDoncila** [comment]
Closed by #734 and #745.
